lib/fuse_lowlevel.c

```diff
@@ -2658,9 +2658,6 @@ void fuse_session_process_buf_int(struct fuse_session *se,
 	int res;
 
 	if (buf->flags & FUSE_BUF_IS_FD) {
-		if (buf->size < tmpbuf.buf[0].size)
```

Comment:
How do you know this condition will never be fulfilled? `buf->size` depends on the caller, and `tmpbuf[0].size` has size `write_header_size` (which is independent of `buf->size`).

Reply:
> How do you know this condition will never be fulfilled? `buf->size` depends on the caller, and `tmpbuf[0].size` has size `write_header_size` (which is independent of `buf->size`).

I had thought buf->size would only be assigned by fuse_session_receive_buf_int, which is the only caller,  then if I understand correctly, buf->size will be greater than "sizeof(struct fuse_in_header) + sizeof(struct fuse_write_in) + pagesize" if  FUSE_BUF_IS_FD is flagged.

Reply:
This might well be true, yes. However, making such assumptions is not a good idea. This is because if someone changes the code of `fuse_session_receive_buf_int` in the future, they are unlikely to be aware that `fuse_session_process_buf_int` cannot receive an arbitrary `buf` parameter. The same goes if someone wants to call `fuse_session_process_buf_int` in a new place.